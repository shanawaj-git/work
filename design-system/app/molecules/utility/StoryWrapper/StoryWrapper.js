import React, { useState } from 'react';

import PropTypes from 'prop-types';

import { ThemeProvider } from 'styled-components';
import { ThemeProvider as MuiThemeProvider } from '@mui/material/styles';
import { StylesProvider } from '@mui/styles';
import { getTheme } from 'themes';
import {
  createMaterialUiTheme,
  GlobalAppStyles,
} from 'styledComponents/styles';
import 'tailwind.css';
import { GlobalTheme } from '../../../utils/globalTheme';

const StoryWrapper = props => {
  const { children, themeVariant = 'DEFAULT' } = props;

  GlobalTheme.setTheme(themeVariant);

  const [{ theme }] = useState(getTheme());
  const [muiTheme] = useState(createMaterialUiTheme(theme));
  return (
    <StylesProvider injectFirst>
      <ThemeProvider theme={theme}>
        <GlobalAppStyles />
        <MuiThemeProvider theme={muiTheme}>{children}</MuiThemeProvider>
      </ThemeProvider>
    </StylesProvider>
  );
};

StoryWrapper.propTypes = {
  children: PropTypes.element.isRequired,
  themeVariant: PropTypes.string,
};

export default StoryWrapper;
