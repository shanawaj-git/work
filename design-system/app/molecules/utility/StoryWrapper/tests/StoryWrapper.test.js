import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import renderer from 'react-test-renderer';

import StoryWrapper from '../index';
const shallowRenderer = new ShallowRenderer();

const Story = () => <div>Story</div>;

describe('StoryWrapper.js', () => {
  describe('index', () => {
    it('Should import succesfully', () => {
      const tree = renderer.create(<StoryWrapper>{Story()} </StoryWrapper>);
      expect(tree).toBeTruthy();
    });

    it('Should import story embedded in the themes', () => {
      shallowRenderer.render(<StoryWrapper>{Story()} </StoryWrapper>);
      const renderedOutput = shallowRenderer.getRenderOutput();

      expect(renderedOutput).toMatchSnapshot();
    });
  });
});
