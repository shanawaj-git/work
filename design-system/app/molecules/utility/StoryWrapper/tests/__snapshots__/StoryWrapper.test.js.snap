// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryWrapper.js index Should import story embedded in the themes 1`] = `
<StylesProvider
  injectFirst={true}
>
  <ThemeProvider
    theme={
      Object {
        "border": Object {
          "level1": "0.3px solid rgba(130, 130, 130, 0.5)",
        },
        "boxShadow": Object {
          "focus": Array [
            "box-shadow:0px 0px 10px ",
            "#AAE4FB",
            ";",
          ],
          "level1": Array [
            "box-shadow:0px 3px 10px rgba(0,0,0,0.07);",
          ],
          "level2": Array [
            "box-shadow:0px 6px 10px rgba(0,0,0,0.07);",
          ],
          "level3": Array [
            "box-shadow:0px 10px 20px rgba(0,0,0,0.07);",
          ],
          "level4": Array [
            "box-shadow:0px 14px 28px rgba(0,0,0,0.1);",
          ],
          "level5": Array [
            "box-shadow:0px 19px 38px rgba(0,0,0,0.15);",
          ],
        },
        "boxShadowFocus": Array [
          "&:focus{box-shadow:0px 0px 10px ",
          "#AAE4FB",
          ";}",
        ],
        "breakpoint": Object {
          "lg": 1200,
          "md": 768,
          "sm": 0,
          "xs": 0,
        },
        "color": Object {
          "a100": "#FFE0BC",
          "a200": "#FFC889",
          "a300": "#F5AB56",
          "a400": "#F99727",
          "b100": "#C4F8CC",
          "b200": "#93E8A1",
          "b300": "#70D280",
          "b400": "#44C058",
          "black": "#000000",
          "c100": "#E25F6D",
          "c200": "#FFFFFF",
          "c300": "#E8E8E6",
          "c400": "#9A9D9D",
          "c500": "#495151",
          "c600": "#d4d9d9",
          "d100": "#AAE4FB",
          "d200": "#87D4F3",
          "d300": "#5EC0E7",
          "d400": "#3FA9D3",
          "error1": "#FFE4E4",
          "error2": "#FD9490",
          "error3": "#FA605A",
          "error4": "#ED2C25",
          "font-tertiary": "#3C3C43CC",
          "fontPrimary": "#3A3A3A",
          "fontSecondary": "#9A9D9D",
          "g100": "#F6F6F8",
          "g200": "#DFE4ED",
          "g300": "#D0D0D8",
          "g400": "#7B7B8E",
          "g500": "#13171C",
          "g600": "#F2F2F7",
          "p100": "#F6FAFB",
          "p200": "#FCF7F7",
          "p300": "#E25F6D",
          "p400": "#535858",
          "p500": "#020A0A",
          "white": "#FFFFFF",
        },
        "fontWeight": Object {
          "bold": Array [
            "font-weight:700;",
          ],
          "book": Array [
            "font-weight:400;",
          ],
          "demi": Array [
            "font-weight:600;",
          ],
          "medium": Array [
            "font-weight:500;",
          ],
        },
        "layer": Object {
          "hidden": -1,
          "mobileSocialFooter": 70,
          "modal": 100,
          "navbar": 50,
          "newQuoteButton": 45,
          "popup": 90,
          "select": 5,
          "slideOut": 1101,
          "tooltip": 60,
          "welcomeModal": 1301,
        },
        "overrides": Object {
          "button": Object {
            "borderRadius": "32px",
          },
        },
        "query": Object {
          "desktop": "(min-width: 1200px)",
          "dpr2x": "(min-resolution: 2dppx), (min-resolution: 192dpi)",
          "max": Object {
            "lg": "(max-width: 1199.98px)",
            "md": "(max-width: 767.98px)",
            "sm": "(max-width: -0.02px)",
            "solve": [Function],
            "tablet": "(max-width: 1199.98px)",
            "xs": "(max-width: -0.02px)",
          },
          "min": Object {
            "lg": "(min-width: 1200px)",
            "md": "(min-width: 768px)",
            "sm": "(min-width: 0px)",
            "solve": [Function],
            "tablet": "(min-width: 768px)",
            "xs": "(min-width: 0px)",
          },
          "mobile": "(max-width: 767.98px)",
          "mobileDpr2x": "(max-width: 767.98px) and (min-resolution: 2dppx), (max-width: 767.98px) and (min-resolution: 192dpi)",
          "tablet": "(min-width: 768px) and (max-width: 1199.98px)",
        },
        "typo": Object {
          "bodyL": Array [
            "font-size:",
            "20px",
            ";font-family:",
            "Poppins",
            ";line-height:",
            "150%",
            ";",
          ],
          "bodyM": Array [
            "font-size:",
            "16px",
            ";font-family:",
            "Poppins",
            ";line-height:",
            "150%",
            ";",
          ],
          "bodyS": Array [
            "font-size:",
            "14px",
            ";font-family:",
            "Poppins",
            ";line-height:",
            "150%",
            ";",
          ],
          "bodyXS": Array [
            "font-size:",
            "12px",
            ";font-family:",
            "Poppins",
            ";line-height:",
            "150%",
            ";",
          ],
          "displayL": Array [
            "font-size:",
            "36px",
            ";font-family:",
            "Poppins",
            ";line-height:",
            "150%",
            ";",
          ],
          "displayM": Array [
            "font-size:",
            "26px",
            ";font-family:",
            "Poppins",
            ";line-height:",
            "150%",
            ";",
          ],
          "displayXL": Array [
            "font-size:",
            "48px",
            ";font-family:",
            "Poppins",
            ";line-height:",
            "150%",
            ";",
          ],
          "displayXXL": Array [
            "font-size:",
            "64px",
            ";font-family:",
            "Poppins",
            ";line-height:",
            "150%",
            ";",
          ],
        },
      }
    }
  >
    <GlobalStyleComponent />
    <ThemeProvider
      theme={
        Object {
          "breakpoints": Object {
            "between": [Function],
            "down": [Function],
            "keys": Array [
              "xs",
              "sm",
              "md",
              "lg",
            ],
            "not": [Function],
            "only": [Function],
            "unit": "px",
            "up": [Function],
            "values": Object {
              "lg": 1200,
              "md": 768,
              "sm": 0,
              "xs": 0,
            },
          },
          "components": Object {},
          "direction": "ltr",
          "mixins": Object {
            "toolbar": Object {
              "@media (min-width:0px)": Object {
                "minHeight": 64,
              },
              "@media (min-width:0px) and (orientation: landscape)": Object {
                "minHeight": 48,
              },
              "minHeight": 56,
            },
          },
          "overrides": Object {
            "MuiPickersDay": Object {
              "daySelected": Object {
                "color": "#FFFFFF",
              },
            },
            "MuiPickersToolbarText": Object {
              "toolbarBtnSelected": Object {
                "color": "#FFFFFF",
              },
              "toolbarTxt": Object {
                "color": "#FFFFFF",
              },
            },
          },
          "palette": Object {
            "action": Object {
              "activatedOpacity": 0.12,
              "active": "rgba(0, 0, 0, 0.54)",
              "disabled": "rgba(0, 0, 0, 0.26)",
              "disabledBackground": "rgba(0, 0, 0, 0.12)",
              "disabledOpacity": 0.38,
              "focus": "rgba(0, 0, 0, 0.12)",
              "focusOpacity": 0.12,
              "hover": "rgba(0, 0, 0, 0.04)",
              "hoverOpacity": 0.04,
              "selected": "rgba(0, 0, 0, 0.08)",
              "selectedOpacity": 0.08,
            },
            "augmentColor": [Function],
            "background": Object {
              "default": "#F6F6F8",
              "paper": "#fff",
            },
            "common": Object {
              "black": "#000",
              "white": "#fff",
            },
            "contrastThreshold": 3,
            "divider": "rgba(0, 0, 0, 0.12)",
            "error": Object {
              "contrastText": "#ED2C25",
              "dark": "#ED2C25",
              "light": "#ED2C25",
              "main": "#ED2C25",
            },
            "getContrastText": [Function],
            "grey": Object {
              "100": "#f5f5f5",
              "200": "#eeeeee",
              "300": "#e0e0e0",
              "400": "#bdbdbd",
              "50": "#fafafa",
              "500": "#9e9e9e",
              "600": "#757575",
              "700": "#616161",
              "800": "#424242",
              "900": "#212121",
              "A100": "#f5f5f5",
              "A200": "#eeeeee",
              "A400": "#bdbdbd",
              "A700": "#616161",
            },
            "info": Object {
              "contrastText": "#fff",
              "dark": "#01579b",
              "light": "#03a9f4",
              "main": "#0288d1",
            },
            "mode": "light",
            "primary": Object {
              "contrastText": "#FFFFFF",
              "dark": "#E25F6D",
              "light": "#E25F6D",
              "main": "#E25F6D",
            },
            "secondary": Object {
              "contrastText": "#fff",
              "dark": "#7b1fa2",
              "light": "#ba68c8",
              "main": "#9c27b0",
            },
            "success": Object {
              "contrastText": "#fff",
              "dark": "#1b5e20",
              "light": "#4caf50",
              "main": "#2e7d32",
            },
            "text": Object {
              "disabled": "rgba(0, 0, 0, 0.38)",
              "primary": "rgba(0, 0, 0, 0.87)",
              "secondary": "rgba(0, 0, 0, 0.6)",
            },
            "tonalOffset": 0.2,
            "warning": Object {
              "contrastText": "#fff",
              "dark": "#e65100",
              "light": "#ff9800",
              "main": "#ed6c02",
            },
          },
          "shadows": Array [
            "none",
            "0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12)",
            "0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12)",
            "0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12)",
            "0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12)",
            "0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12)",
            "0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12)",
            "0px 4px 5px -2px rgba(0,0,0,0.2),0px 7px 10px 1px rgba(0,0,0,0.14),0px 2px 16px 1px rgba(0,0,0,0.12)",
            "0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12)",
            "0px 5px 6px -3px rgba(0,0,0,0.2),0px 9px 12px 1px rgba(0,0,0,0.14),0px 3px 16px 2px rgba(0,0,0,0.12)",
            "0px 6px 6px -3px rgba(0,0,0,0.2),0px 10px 14px 1px rgba(0,0,0,0.14),0px 4px 18px 3px rgba(0,0,0,0.12)",
            "0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)",
            "0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12)",
            "0px 7px 8px -4px rgba(0,0,0,0.2),0px 13px 19px 2px rgba(0,0,0,0.14),0px 5px 24px 4px rgba(0,0,0,0.12)",
            "0px 7px 9px -4px rgba(0,0,0,0.2),0px 14px 21px 2px rgba(0,0,0,0.14),0px 5px 26px 4px rgba(0,0,0,0.12)",
            "0px 8px 9px -5px rgba(0,0,0,0.2),0px 15px 22px 2px rgba(0,0,0,0.14),0px 6px 28px 5px rgba(0,0,0,0.12)",
            "0px 8px 10px -5px rgba(0,0,0,0.2),0px 16px 24px 2px rgba(0,0,0,0.14),0px 6px 30px 5px rgba(0,0,0,0.12)",
            "0px 8px 11px -5px rgba(0,0,0,0.2),0px 17px 26px 2px rgba(0,0,0,0.14),0px 6px 32px 5px rgba(0,0,0,0.12)",
            "0px 9px 11px -5px rgba(0,0,0,0.2),0px 18px 28px 2px rgba(0,0,0,0.14),0px 7px 34px 6px rgba(0,0,0,0.12)",
            "0px 9px 12px -6px rgba(0,0,0,0.2),0px 19px 29px 2px rgba(0,0,0,0.14),0px 7px 36px 6px rgba(0,0,0,0.12)",
            "0px 10px 13px -6px rgba(0,0,0,0.2),0px 20px 31px 3px rgba(0,0,0,0.14),0px 8px 38px 7px rgba(0,0,0,0.12)",
            "0px 10px 13px -6px rgba(0,0,0,0.2),0px 21px 33px 3px rgba(0,0,0,0.14),0px 8px 40px 7px rgba(0,0,0,0.12)",
            "0px 10px 14px -6px rgba(0,0,0,0.2),0px 22px 35px 3px rgba(0,0,0,0.14),0px 8px 42px 7px rgba(0,0,0,0.12)",
            "0px 11px 14px -7px rgba(0,0,0,0.2),0px 23px 36px 3px rgba(0,0,0,0.14),0px 9px 44px 8px rgba(0,0,0,0.12)",
            "0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12)",
          ],
          "shape": Object {
            "borderRadius": 4,
          },
          "spacing": [Function],
          "transitions": Object {
            "create": [Function],
            "duration": Object {
              "complex": 375,
              "enteringScreen": 225,
              "leavingScreen": 195,
              "short": 250,
              "shorter": 200,
              "shortest": 150,
              "standard": 300,
            },
            "easing": Object {
              "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
              "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
              "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
              "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
            },
            "getAutoHeightDuration": [Function],
          },
          "typography": Object {
            "body1": Object {
              "fontFamily": "Poppins",
              "fontSize": "1rem",
              "fontWeight": 400,
              "lineHeight": 1.5,
            },
            "body2": Object {
              "fontFamily": "Poppins",
              "fontSize": "0.875rem",
              "fontWeight": 400,
              "lineHeight": 1.43,
            },
            "button": Object {
              "fontFamily": "Poppins",
              "fontSize": "0.875rem",
              "fontWeight": 500,
              "lineHeight": 1.75,
              "textTransform": "uppercase",
            },
            "caption": Object {
              "fontFamily": "Poppins",
              "fontSize": "0.75rem",
              "fontWeight": 400,
              "lineHeight": 1.66,
            },
            "fontFamily": "Poppins",
            "fontSize": 14,
            "fontWeightBold": 700,
            "fontWeightLight": 300,
            "fontWeightMedium": 500,
            "fontWeightRegular": 400,
            "h1": Object {
              "fontFamily": "Poppins",
              "fontSize": "6rem",
              "fontWeight": 300,
              "lineHeight": 1.167,
            },
            "h2": Object {
              "fontFamily": "Poppins",
              "fontSize": "3.75rem",
              "fontWeight": 300,
              "lineHeight": 1.2,
            },
            "h3": Object {
              "fontFamily": "Poppins",
              "fontSize": "3rem",
              "fontWeight": 400,
              "lineHeight": 1.167,
            },
            "h4": Object {
              "fontFamily": "Poppins",
              "fontSize": "2.125rem",
              "fontWeight": 400,
              "lineHeight": 1.235,
            },
            "h5": Object {
              "fontFamily": "Poppins",
              "fontSize": "1.5rem",
              "fontWeight": 400,
              "lineHeight": 1.334,
            },
            "h6": Object {
              "fontFamily": "Poppins",
              "fontSize": "1.25rem",
              "fontWeight": 500,
              "lineHeight": 1.6,
            },
            "htmlFontSize": 16,
            "overline": Object {
              "fontFamily": "Poppins",
              "fontSize": "0.75rem",
              "fontWeight": 400,
              "lineHeight": 2.66,
              "textTransform": "uppercase",
            },
            "pxToRem": [Function],
            "subtitle1": Object {
              "fontFamily": "Poppins",
              "fontSize": "1rem",
              "fontWeight": 400,
              "lineHeight": 1.75,
            },
            "subtitle2": Object {
              "fontFamily": "Poppins",
              "fontSize": "0.875rem",
              "fontWeight": 500,
              "lineHeight": 1.57,
            },
            "useNextVariants": true,
          },
          "zIndex": Object {
            "appBar": 1100,
            "drawer": 1200,
            "mobileStepper": 1000,
            "modal": 1300,
            "snackbar": 1400,
            "speedDial": 1050,
            "tooltip": 1500,
          },
        }
      }
    >
      <div>
        Story
      </div>
       
    </ThemeProvider>
  </ThemeProvider>
</StylesProvider>
`;
